core/src/main/java/com/amplifyframework/auth/AuthException.java

```diff
@@ -191,4 +191,384 @@ public UnknownException(Throwable cause) {
+            }
+        }
+
+        /**
+         * Could not perform the action because Amazon Cognito service encounters an invalid AWS Lambda response
+         * or encounters an unexpected exception with the AWS Lambda service.
+         */
+        public static class LambdaException extends AWSCognitoAuthException {
+            private static final long serialVersionUID = 1L;
+            private static final String MESSAGE =
+                    "Amazon Cognito service encounters an invalid AWS Lambda response " +
+                            "or encounters an unexpected exception with the AWS Lambda service.";
+            private static final String RECOVERY_SUGGESTION = "Turn off and re-attempt this operation.";
+
+            /**
+             * Default message/recovery suggestion with a cause.
+             * @param cause The original error.
+             */
+            public LambdaException(Throwable cause) {
+                super(MESSAGE, cause, RECOVERY_SUGGESTION);
+            }
+        }
+
+        /**
+         * Could not perform the action because device is not tracked.
+         */
+        public static class DeviceNotTrackedException extends AWSCognitoAuthException {
+            private static final long serialVersionUID = 1L;
+            private static final String MESSAGE = "Device is not tracked.";
+            private static final String RECOVERY_SUGGESTION = "Turn off and re-attempt this operation.";
+
+            /**
+             * Default message/recovery suggestion with a cause.
+             * @param cause The original error.
+             */
+            public DeviceNotTrackedException(Throwable cause) {
+                super(MESSAGE, cause, RECOVERY_SUGGESTION);
+            }
+        }
+
+        /**
+         * Could not perform the action because there exists error in loading web UI.
+         */
+        public static class ErrorLoadingUIException extends AWSCognitoAuthException {
+            private static final long serialVersionUID = 1L;
+            private static final String MESSAGE = "Error in loading the web UI.";
+            private static final String RECOVERY_SUGGESTION = "Turn off and re-attempt this operation.";
+
+            /**
+             * Default message/recovery suggestion with a cause.
+             * @param cause The original error.
+             */
+            public ErrorLoadingUIException(Throwable cause) {
+                super(MESSAGE, cause, RECOVERY_SUGGESTION);
+            }
+        }
+
+        /**
+         * Could not perform the action because user cancelled the step.
+         */
+        public static class UserCancelledException extends AWSCognitoAuthException {
+            private static final long serialVersionUID = 1L;
+            private static final String MESSAGE = "User cancelled the step.";
+            private static final String RECOVERY_SUGGESTION = "Turn off and re-attempt this operation.";
```

Comment:
"You can provide an option to retry the operation to the user but should not automatically do it as the user has expressed they wanted to cancel it."